packages/react-native/Libraries/LinkingIOS/RCTLinkingManager.mm

```diff
@@ -81,6 +81,52 @@ + (BOOL)application:(UIApplication *)application
   return YES;
 }
 
+// iOS 13+ - Corresponding apis for SceneDelegate
+
+// Both Deep Links and Universal Links - called when the app was previously NOT running
++ (BOOL)scene:(UIScene *)scene
+    willConnectTo:(UISceneSession *)session
+      options:(UISceneConnectionOptions *) connectionOptions
+{
+
+  // Deep Links
+  for (UIOpenURLContext *urlContext in connectionOptions.URLContexts) {
+    if (urlContext) {
+      postNotificationWithURL(urlContext.URL, self);
+    }
+  }
+
+  // Universal Links
+  for (NSUserActivity *userActivity in connectionOptions.userActivities) {
+    if ([userActivity.activityType isEqualToString:NSUserActivityTypeBrowsingWeb]) {
+      postNotificationWithURL(userActivity.webpageURL, self);
+    }
+  }
```

Comment:
This should work, but the problem lies deeper because `connectionOptions` will come empty in the first place, due to the root View being started without `launchOptions`